### ISSUE TITLE
linter: panic in `jest/no_conditional_expect`

### ISSUE THREAD

**Issue by Boshen** [oxc-project/oxc#3325]
Found in https://github.com/oxc-project/oxlint-ecosystem-ci/actions/runs/9125633095/job/25092301842

```javascript
/** @type {(vnode: import('../../src/internal').VNode) => void} */
function verifyVNodeTree(vnode) {
	if (vnode._nextDom) {
		expect.fail('vnode should not have _nextDom:' + vnode._nextDom);
	}

	if (vnode._children) {
		for (let child of vnode._children) {
			if (child) {
				verifyVNodeTree(child);
			}
		}
	}
}
```
